frontend/src/components/core/StreamlitDialog/ThemeCreator.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { ReactElement } from "react"
+import { toHex } from "color2k"
+import humanizeString from "humanize-string"
+import { CustomThemeConfig } from "autogen/proto"
+import Button, { Kind } from "components/shared/Button"
+import UISelectbox from "components/shared/Dropdown"
+import { fonts } from "theme/primitives/typography"
+import {
+  StyledButtonContainer,
+  StyledHeader,
+  StyledSmall,
+  StyledThemeColorPicker,
+  StyledThemeCreator,
+  StyledThemeCreatorWrapper,
+  StyledThemeDesc,
+} from "./styled-components"
+
+export interface Props {
+  label?: string
+  themeInput: Partial<CustomThemeConfig>
+  updateThemeInput: (themeInput: Partial<CustomThemeConfig>) => any
+}
+
+const themeBuilder: any = {
```

Comment:
nit: I think it'd be worth typing this a bit more thoroughly since inner objects have more or less the same shape with only the `options` property being optional